service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/ainotebook/ControlledAiNotebookHandler.java

```diff
@@ -50,16 +63,53 @@ public WsmResource makeResourceFromDb(DbResource dbResource) {
         Optional.ofNullable(attributes.getProjectId())
             .orElse(gcpCloudContextService.getRequiredGcpProject(dbResource.getWorkspaceId()));
 
-    var resource =
-        ControlledAiNotebookInstanceResource.builder()
-            .common(new ControlledResourceFields(dbResource))
-            .instanceId(attributes.getInstanceId())
-            .location(attributes.getLocation())
-            .projectId(projectId)
-            .machineType(attributes.getMachineType())
-            .acceleratorConfig(attributes.getAcceleratorConfig())
-            .build();
-    return resource;
+    // Notebook attributes created prior to PF-2323 do not have machineType or acceleratorConfig
+    // stored. Note: machineType is required.
+    // Thus, retrieve these two values from the cloud.
+    String retrievedMachineType = attributes.getMachineType();
+    AcceleratorConfig retrievedAcceleratorConfig = attributes.getAcceleratorConfig();
+
+    if (retrievedMachineType == null) {
+      try (NotebookServiceClient notebookServiceClient = NotebookServiceClient.create()) {
```

Comment:
@ddietterich Do we backfill attributes as we read the notebooks? Or do we want a separate process akin to the BQ backfill?

Reply:
I vote for the BQ-backfill type of process

Reply:
I think that covers it well